src/Text/Pandoc/Readers/RST.hs

```diff
@@ -770,6 +771,26 @@ tableDirective top _fields body = do
                                   aligns' widths' header' rows'
        _ -> return mempty
 
+listTableDirective :: PandocMonad m => String -> [(String, String)] -> String -> RSTParser m Blocks
+listTableDirective top _fields body = do
+  bs <- parseFromString parseBlocks body
+  title <- do { parseFromString (trimInlines . mconcat <$> many inline) top }
+  let rows = takeRows $ B.toList bs
+      numOfCols = length $ rows !! 0
+      headerRows = case lookup "header-rows" _fields of
+                     Just x -> read x :: Int
```

Comment:
`read` is also partial and will crash at runtime if `x` can't be read as an `Int`. We don't want that.  Check out `safeRead` from `Text.Pandoc.Shared`.